File: blog-backend/blog-pojo/src/main/java/com/yvmoux/blog/entity/UserLike.java
package com.yvmoux.blog.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("user_like")
public class UserLike {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private Long articleId;

    private LocalDateTime createdAt;
}

File: blog-backend-go/internal/model/user_like.go
package model

import "time"

// UserLike maps to the `user_like` table.
type UserLike struct {
	ID        int64     `gorm:"column:id;primaryKey;autoIncrement" json:"id"`
	UserID    int64     `gorm:"column:user_id;not null" json:"userId"`
	ArticleID int64     `gorm:"column:article_id;not null" json:"articleId"`
	CreatedAt time.Time `gorm:"column:created_at;not null" json:"createdAt"`
}

func (UserLike) TableName() string {
	return "user_like"
}
